docs/rules/naming-convention.md

```diff
@@ -14,151 +14,120 @@ Require names to follow specified conventions.
 ### Incorrect
 
 ```graphql
-# eslint @graphql-eslint/naming-convention: ['error', { ObjectTypeDefinition: 'PascalCase' }]
+# eslint @graphql-eslint/naming-convention: ['error', { types: 'PascalCase', fields: 'camelCase' }]
 
-type someTypeName {
-  f: String!
+type user {
+  first_name: String!
 }
 ```
 
 ### Correct
 
 ```graphql
-# eslint @graphql-eslint/naming-convention: ['error', { FieldDefinition: 'camelCase', ObjectTypeDefinition: 'PascalCase' }]
+# eslint @graphql-eslint/naming-convention: ['error', { types: 'PascalCase', fields: 'camelCase' }]
 
-type SomeTypeName {
-  someFieldName: String
+type User {
+  firstName: String
 }
 ```
 
 ## Config Schema
 
 The schema defines the following properties:
 
-### `FieldDefinition`
+### `types`
 
-The object must be one of the following types:
-
-* `asString`
-* `asObject`
+Includes:
 
-### `InputObjectTypeDefinition`
+- `ObjectTypeDefinition`
+- `InterfaceTypeDefinition`
+- `EnumTypeDefinition`
+- `ScalarTypeDefinition`
+- `InputObjectTypeDefinition`
+- `UnionTypeDefinition`
 
 The object must be one of the following types:
 
 * `asString`
 * `asObject`
 
-### `EnumValueDefinition`
+### `fields`
 
-The object must be one of the following types:
-
-* `asString`
-* `asObject`
-
-### `InputValueDefinition`
-
-The object must be one of the following types:
-
-* `asString`
-* `asObject`
+Includes:
 
-### `ObjectTypeDefinition`
+- `FieldDefinition`
+- `InputValueDefinition`
+- `VariableDefinition`
+- `Argument`
+- `DirectiveDefinition`
 
 The object must be one of the following types:
 
 * `asString`
```

Comment:
I know these already exist. I found them unclear though. Any reason we are prefixing with `as` rather than just saying "String"? Is that an ESLint thing?

Reply:
Nope, it's not an ESLint thing, it's just a name of subschema definition.